contracts/upgradeable_contracts/erc20_to_native/ForeignBridgeErcToNative.sol

```diff
@@ -83,23 +97,113 @@ contract ForeignBridgeErcToNative is BasicForeignBridge, ERC20Bridge, OtherSideB
         super._relayTokens(_sender, _receiver, _amount);
     }
 
-    function migrateToMCD(address _migrationContract) external onlyOwner {
+    function migrateToMCD() external onlyOwner {
         bytes32 storageAddress = 0x3378953eb16363e06fd9ea9701d36ed7285d206d9de7df55b778462d74596a89; // keccak256(abi.encodePacked("migrationToMcdCompleted"))
-
         require(!boolStorage[storageAddress]);
-        require(AddressUtils.isContract(_migrationContract));
-
-        uint256 curBalance = erc20token().balanceOf(address(this));
-        require(erc20token().approve(_migrationContract, curBalance));
-        //It is important to note that this action will cause appearing of `Transfer`
-        //event as part of the tokens minting
-        IScdMcdMigration(_migrationContract).swapSaiToDai(curBalance);
-        address saiContract = erc20token();
-        address mcdContract = IDaiAdapter(IScdMcdMigration(_migrationContract).daiJoin()).dai();
+
+        address mcdContract = IDaiAdapter(migrationContract().daiJoin()).dai();
         setErc20token(mcdContract);
-        require(erc20token().balanceOf(address(this)) == curBalance);
 
-        emit TokensSwapped(saiContract, erc20token(), curBalance);
+        uintStorage[MIN_HDTOKEN_BALANCE] = 10 ether;
+
+        swapTokens();
+
         boolStorage[storageAddress] = true;
     }
+
+    function saiTopContract() internal pure returns (ISaiTop) {
+        return ISaiTop(0x9b0ccf7C8994E19F39b2B4CF708e0A7DF65fA8a3);
+    }
+
+    function isTokenSwapAllowed(uint256 _ts) public view returns (bool) {
+        uint256 esTs = saiTopContract().caged();
+        if (esTs > 0 && _ts > esTs) {
+            return false;
+        }
+        return true;
+    }
+
+    function halfDuplexErc20token() public pure returns (ERC20) {
+        return ERC20(0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359);
+    }
+
+    function setMinHDTokenBalance(uint256 _minBalance) external onlyOwner {
+        uintStorage[MIN_HDTOKEN_BALANCE] = _minBalance;
+    }
+
+    function minHDTokenBalance() public view returns (uint256) {
+        return uintStorage[MIN_HDTOKEN_BALANCE];
+    }
+
+    function isHDTokenBalanceAboveMinBalance() public view returns (bool) {
+        if (tokenBalance(halfDuplexErc20token()) > minHDTokenBalance()) {
+            return true;
+        }
+        return false;
+    }
+
+    function tokenBalance(ERC20 _token) internal view returns (uint256) {
+        return _token.balanceOf(address(this));
+    }
+
+    function migrationContract() internal pure returns (IScdMcdMigration) {
+        return IScdMcdMigration(0xc73e0383F3Aff3215E6f04B0331D58CeCf0Ab849);
+    }
+
+    function swapTokens() public {
```

Comment:
should we increase the version returned by `getBridgeInterfacesVersion`?

Reply:
Yes, it makes sense. Updated to `2.5.0` https://github.com/poanetwork/tokenbridge-contracts/pull/319/commits/c2c2c4cc2b5308b1fbe5a7bdd872d5238e297636